hta/analyzers/trace_counters.py

```diff
@@ -98,14 +102,15 @@ def get_queue_length_time_series(
             "stays constant until the next update."
         )
 
-        return {rank: TraceCounters._get_queue_length_time_series_for_rank(t, rank) for rank in ranks}
+        result = {rank: TraceCounters._get_queue_length_time_series_for_rank(t, rank) for rank in ranks}
+        return dict(filter(lambda x: x[1] is not None, result.items()))
 
     @classmethod
     def get_queue_length_summary(
         cls,
         t: "Trace",
         ranks: Optional[List[int]] = None,
-    ) -> pd.DataFrame:
+    ) -> Optional[pd.DataFrame]:
         """
         Returns a dataframe with queue length statistics per CUDA stream and rank.
```

Comment:
@briancoutinho please edit the docstring to reflect that the return type is Optional[pd.Datafame] and not a pd.Dataframe.